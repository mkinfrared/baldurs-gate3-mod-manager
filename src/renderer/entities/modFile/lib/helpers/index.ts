export { extractPakFile } from "./extractPakFile";

export { getModFiles } from "./getModFiles";

export { getModInfo } from "./getModInfo";

export { readZipInfo } from "./readZipInfo";
