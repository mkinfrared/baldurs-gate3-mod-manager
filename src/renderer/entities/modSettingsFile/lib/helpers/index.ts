export { activateMod } from "./activateMod";

export { deactivateMod } from "./deactivateMod";

export { getCurrentSettings } from "./getCurrentSettings";

export { getInstalledMods } from "./getInstalledMods";

export { getModInfo } from "./getModInfo";

export { getNodeValue } from "./getNodeValue";
