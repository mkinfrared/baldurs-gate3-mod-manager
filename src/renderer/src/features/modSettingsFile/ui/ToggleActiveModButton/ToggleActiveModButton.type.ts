export type ToggleActiveModButtonProps = {
  /**
   * a string that will be applied as a css class to parent element
   */
  className?: string;
  fileName: string;
  isActive: boolean;
  uuid: string | undefined;
};
