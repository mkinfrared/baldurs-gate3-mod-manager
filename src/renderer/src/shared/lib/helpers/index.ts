export { classNames } from "./classNames";

export { compose } from "./compose";

export { delay } from "./delay";

export { getNodeText } from "./getNodeText";

export { lazyImport } from "./lazyImport";

export { mergeRefs } from "./mergeRefs";

export { trpcClient, trpc } from "./trpc";

export * from "./typeGuards";
