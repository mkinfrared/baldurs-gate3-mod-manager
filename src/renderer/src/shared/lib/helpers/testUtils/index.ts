export { renderWithRouter, withQueryProvider } from "./testUtils";
