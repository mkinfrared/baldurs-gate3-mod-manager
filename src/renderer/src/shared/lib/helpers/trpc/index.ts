export { trpc, trpcClient } from "./trpc";
