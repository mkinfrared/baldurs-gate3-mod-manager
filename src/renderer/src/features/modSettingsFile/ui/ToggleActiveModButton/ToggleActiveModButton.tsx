import { classNames, trpc } from "@renderer/shared/lib/helpers";
import { Button } from "@renderer/shared/ui";

import css from "./ToggleActiveModButton.module.scss";
import { ToggleActiveModButtonProps } from "./ToggleActiveModButton.type";

const ToggleActiveModButton = ({
  className,
  fileName,
  game,
  isActive,
  uuid,
}: ToggleActiveModButtonProps) => {
  const utils = trpc.useContext();

  const { mutateAsync, isLoading } = trpc.mod.toggleActiveMod.useMutation({
    onSuccess: () => {
      utils.mod.getInstalledMods.invalidate();
    },
  });

  if (!uuid) {
    return <div />;
  }

  const handleClick = () => {
    mutateAsync({ file: fileName, gameKey: game });
  };

  return (
    <Button
      className={classNames(css.ToggleActiveModButton, className)}
      loading={isLoading}
      disabled={isLoading}
      onClick={handleClick}
      color={isActive ? "secondary" : "primary"}
    >
      {isActive ? "Disable" : "Enable"}
    </Button>
  );
};

export { ToggleActiveModButton };
