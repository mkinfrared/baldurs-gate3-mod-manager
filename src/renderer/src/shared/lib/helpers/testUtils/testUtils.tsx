/* eslint-disable import/no-unused-modules */
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { render } from "@testing-library/react";
import { userEvent } from "@testing-library/user-event";
import { ReactElement, ReactNode } from "react";
import { BrowserRouter, MemoryRouter } from "react-router-dom";

import { trpc, trpcClient } from "@renderer/shared/lib/helpers/trpc";

// https://testing-library.com/docs/example-react-router/
const renderWithRouter = (ui: ReactElement, { route = "/" } = {}) => {
  window.history.pushState({}, "Test page", route);

  return {
    user: userEvent.setup(),
    ...render(ui, { wrapper: BrowserRouter }),
  };
};

const queryClient = new QueryClient();

const withQueryProvider = (children: ReactNode) => (
  <trpc.Provider client={trpcClient} queryClient={queryClient}>
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
  </trpc.Provider>
);

const withRouter = (children: ReactNode) => (
  <MemoryRouter>{children}</MemoryRouter>
);

export { renderWithRouter, withQueryProvider, withRouter };
