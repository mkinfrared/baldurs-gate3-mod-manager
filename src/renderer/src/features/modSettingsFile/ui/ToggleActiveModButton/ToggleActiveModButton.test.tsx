import { render } from "@testing-library/react";

import { withQueryProvider } from "@renderer/shared/lib/helpers/testUtils";

import { ToggleActiveModButton } from "./ToggleActiveModButton";

describe("<ToggleActiveModButton />", () => {
  const fileName = "foobar";
  const game = "BG3";
  const isActive = true;
  const uuid = "modUUID";

  const Component = (
    <ToggleActiveModButton
      fileName={fileName}
      game={game}
      isActive={isActive}
      uuid={uuid}
    />
  );

  it("should be defined", () => {
    expect(ToggleActiveModButton).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(withQueryProvider(Component));

    expect(container).toMatchSnapshot();
  });
});
