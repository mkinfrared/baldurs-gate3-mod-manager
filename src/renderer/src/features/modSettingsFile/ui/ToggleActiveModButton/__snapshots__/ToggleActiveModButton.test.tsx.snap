// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ToggleActiveModButton /> > should match the snapshot 1`] = `
<div>
  <button
    class="Button ToggleActiveModButton contained primary"
    data-testid="Button"
    type="button"
  >
    <svg
      class="Spinner spinner"
    >
      bars.svg
    </svg>
    <span
      class="content"
    >
      Deactivate
    </span>
  </button>
</div>
`;
